### ISSUE TITLE
Prioritize search results based on usage

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#6053]
With the new search-to-add workflow, producing highly relevant search results is even more important than before. I've come across many instances where a very common preset is annoyingly below less-common presets. We should factor taginfo usage into our search algorithm.

Service Road should be first:
<img width="399" alt="Screen Shot 2019-03-17 at 12 08 46 PM" src="https://user-images.githubusercontent.com/2046746/54494165-d9ac4a80-48ad-11e9-9c5c-3d96fa2c3c37.png">

Foot Path isn't even shown:
<img width="525" alt="Screen Shot 2019-03-17 at 12 01 32 PM" src="https://user-images.githubusercontent.com/2046746/54494089-1592e000-48ad-11e9-9bec-bd6e1f8b1b1e.png">

**Comment by slhh** [comment]
@quincylvania 
Taginfo seems to be useful to prioritize presets which match to the same extend, but
I think we should avoid to prioritize more frequent ones over better matching ones.

**Comment by bhousel** [comment]
Yep this relates to #2000.. We've known for a while that  "unmodified Levenshtein distance is probably not the best metric here."  I closed it years ago because nobody seemed interested in digging in to improve the algorithm.

I'm also reluctant to give too much weight to taginfo popularity, but I agree it would be useful as another thing to consider in the weight.  We'd need to calculate these in at build-time to do it perfectly, or it might be ok to just do _a few_ taginfo queries when the user's editing session starts to get the most popular things, and give them a slight boost.

Also worth mentioning - you picked "Service" as an example, which might be one of the worst cases because that tag is used for several different things in OSM.  

And it doesn't seem _that_ weird that all the top results for "foo" are "food".

**Comment by TAQ2** [comment]
Is this an issue that could have the help wanted label?

**Comment by 1ec5** [comment]
> We'd need to calculate these in at build-time to do it perfectly

I think it would be fine to bake some rankings (hopefully compressed somehow) into the build. If a tag suddenly and dramatically gains in popularity, it’d be fine for iD’s preset search results to lag that by a month or two. IDEs and IMEs similarly bundle some rudimentary weighting statistics into the build.

Interesting to think about brand presets being weighted based on usage, since I think iD already gets those statistics from NSI.

**Comment by quincylvania** [comment]
> Is this an issue that could have the help wanted label?

@TAQ2 Not necessarily. We don't use that label too often but mostly it's for stuff we don't know how to do or don't want to spend core time on.

**Comment by matkoniecz** [comment]
matching seems a bit smarter nowadays

are there other examples where taking into account raw tag use would help?

<img width="390" height="650" alt="Image" src="https://github.com/user-attachments/assets/8df0c383-b443-4f3d-b7e1-3dee71525058" />
<img width="376" height="512" alt="Image" src="https://github.com/user-attachments/assets/64c9ced6-f9a3-4125-bf20-cf3eaa5304ea" />
